1)

Un KPI (Key Performance Indicator) un est indicateur clé de performance, qui permet de réaliser et présenter des tableaux de bords pour de la gestion. Ils sont utilisés pour évaluer la réussite ou non d'un objectif


2) Import de Pandas et Numpy

In [2]:
import pandas as pd
import numpy as npy

3) Import du fichier 'eval_data.csv'

In [3]:
df = pd.read_csv("csv_files/eval_data.csv")

4) Affichage des 10 premières lignes

In [4]:
df.head(10)

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   
5           5         3         1                           Chicken Bowl   
6           6         3         1                          Side of Chips   
7           7         4         1                          Steak Burrito   
8           8         4         1                       Steak Soft Tacos   
9           9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

5) Afficher la dimension du dataset

In [68]:
df.ndim

2

In [69]:
df.shape

(4622, 6)

6) Afficher et décrire les colonnes du dataset

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Unnamed: 0          4622 non-null   int64 
 1   order_id            4622 non-null   int64 
 2   quantity            4622 non-null   int64 
 3   item_name           4622 non-null   object
 4   choice_description  3376 non-null   object
 5   item_price          4522 non-null   object
dtypes: int64(3), object(3)
memory usage: 216.8+ KB


7) Afficher les statistiques descriptives du dataset

In [7]:
df.describe()

Unnamed: 0     order_id     quantity
count  4622.000000  4622.000000  4622.000000
mean   2310.500000   927.254868     1.075725
std    1334.400802   528.890796     0.410186
min       0.000000     1.000000     1.000000
25%    1155.250000   477.250000     1.000000
50%    2310.500000   926.000000     1.000000
75%    3465.750000  1393.000000     1.000000
max    4621.000000  1834.000000    15.000000

8) Afficher l'article le plus commandé

In [61]:
df.groupby(['item_name'])['quantity'].sum().sort_values(ascending=False).head(1)

item_name
Chicken Bowl    761
Name: quantity, dtype: int64

L'article "Chicken Bowl" a été commandé 761 fois

9) Nombre d'items commandés au total

In [43]:
df["quantity"].sum()

4972

10) Chiffre d'affaire de l'entreprise

In [70]:
df["item_price"].sum()

AttributeError: 'Series' object has no attribute 'to_float'

11) Revenu moyen par commande